Format.pm

```diff
@@ -536,14 +593,124 @@ sub round
 
     # We need to add 1e-14 to avoid some rounding errors due to the
     # way floating point numbers work - see string-eq test in t/round.t
-    $result = int($product + .5 + 1e-14) / $multiplier;
-    $result = -$result if $sign < 0;
+    if (
+        $roundoption == ROUND_FLOOR
+        or ($roundoption == ROUND_ABS_FLOOR and not $sign < 0)
```

Comment:
done, but now it is not consistent with the behavior from the Math::BigFloat part (`not $rounded->is_neg()`